manim/utils/config_utils.py

```diff
@@ -0,0 +1,382 @@
+
+    # The following overrides -w, -a, -g, and -i
+    parser.add_argument(
+        "--dry_run",
+        action="store_true",
+        help="Do a dry run (render scenes but generate no output files)",
+    )
+
+    # The following overrides PNG_MODE, MOVIE_FILE_EXTENSION, and
+    # BACKGROUND_OPACITY
+    parser.add_argument(
+        "-t",
+        "--transparent",
+        action="store_true",
+        help="Render to a movie file with an alpha channel",
+    )
+
+    # The following are mutually exclusive and each overrides
+    # FRAME_RATE, PIXEL_HEIGHT, and PIXEL_WIDTH,
+    parser.add_argument(
+        "-l",
+        "--low_quality",
+        action="store_true",
+        help="Render at low quality (for fastest rendering)",
+    )
+    parser.add_argument(
+        "-m",
+        "--medium_quality",
+        action="store_true",
+        help="Render at medium quality (for much faster rendering)",
+    )
+    parser.add_argument(
+        "-e",
+        "--high_quality",
+        action="store_true",
+        help="Render at high quality (for slightly faster rendering)",
+    )
+    parser.add_argument(
+        "-k",
+        "--fourk_quality",
+        action="store_true",
+        help="Render at 4K quality (slower rendering)",
+    )
+
+    # This overrides any of the above
+    parser.add_argument(
+        "-r", "--resolution", help='Resolution, passed as "height,width"',
+    )
+
+    # This sets FROM_ANIMATION_NUMBER and UPTO_ANIMATION_NUMBER
+    parser.add_argument(
+        "-n",
+        "--from_animation_number",
+        help="Start rendering not from the first animation, but"
+        "from another, specified by its index.  If you pass"
+        'in two comma separated values, e.g. "3,6", it will end'
+        "the rendering at the second value",
+    )
+
+    # Specify the manim.cfg file
+    parser.add_argument(
+        "--config_file", help="Specify the configuration file",
+    )
+
+    return parser.parse_args(arg_list)
+
+
+def _init_dirs(config):
+    # Make sure all folders exist
+    for folder in [
+        config["media_dir"],
+        config["video_dir"],
+        config["tex_dir"],
+        config["text_dir"],
+    ]:
+        if not os.path.exists(folder):
+            os.makedirs(folder)
+
+
+def _from_command_line():
+    """Determine if manim was called from the command line."""
+    # Manim can be called from the command line in three different
+    # ways.  The first two involve using the manim or manimcm commands
+    prog = os.path.split(sys.argv[0])[-1]
+    from_cli_command = prog in ["manim", "manimcm"]
+
+    # The third way involves using `python -m manim ...`.  In this
+    # case, the CLI arguments passed to manim do not include 'manim',
+    # 'manimcm', or even 'python'.  However, the -m flag will always
+    # be the first argument.
+    from_python_m = sys.argv[0] == "-m"
+
+    return from_cli_command or from_python_m
+
+
+def _run_config():
```

Comment:
This function has to be executed twice, once in `logger.py`  and again while importing `config.py` any better idea tha that?
@leotrs